mapedit/studio.cc

```diff
@@ -3164,228 +3160,302 @@ void ExultStudio::set_game_information(
+		0x20, 0x21, 0x22, 0x23, 0x24, 0x25, 0x26, 0x27, 0x28, 0x29, 0x2a, 0x2b, 0x2c, 0x2d, 0x2e, 0x2f, // 2
+		0x30, 0x31, 0x32, 0x33, 0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x3a, 0x3b, 0x3c, 0x3d, 0x3e, 0x3f, // 3
+		0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x48, 0x49, 0x4a, 0x4b, 0x4c, 0x4d, 0x4e, 0x4f, // 4
+		0x50, 0x51, 0x52, 0x53, 0x54, 0x55, 0x56, 0x57, 0x58, 0x59, 0x5a, 0x5b, 0x5c, 0x5d, 0x5e, 0x5f, // 5
+		0x60, 0x61, 0x62, 0x63, 0x64, 0x65, 0x66, 0x67, 0x68, 0x69, 0x6a, 0x6b, 0x6c, 0x6d, 0x6e, 0x6f, // 6
+		0x70, 0x71, 0x72, 0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7a, 0x7b, 0x7c, 0x7d, 0x7e, 0x7f, // 7
+		0x80, 0x81, 0x82, 0x83, 0x84, 0x85, 0x86, 0x87, 0x88, 0x89, 0x8a, 0x8b, 0x8c, 0x8d, 0x8e, 0x8f, // 8
+		0x90, 0x91, 0x92, 0x93, 0x94, 0x95, 0x96, 0x97, 0x98, 0x99, 0x9a, 0x9b, 0x9c, 0x9d, 0x9e, 0x9f, // 9
+		0xa0, 0xa1, 0xa2, 0xa3, 0xa4, 0xa5, 0xa6, 0xa7, 0xa8, 0xa9, 0xaa, 0xab, 0xac, 0xad, 0xae, 0xaf, // a
+		0xb0, 0xb1, 0xb2, 0xb3, 0xb4, 0xb5, 0xb6, 0xb7, 0xb8, 0xb9, 0xba, 0xbb, 0xbc, 0xbd, 0xbe, 0xbf, // b
+		0xc0, 0xc1, 0xc2, 0xc3, 0xc4, 0xc5, 0xc6, 0xc7, 0xc8, 0xc9, 0xca, 0xcb, 0xcc, 0xcd, 0xce, 0xcf, // c
+		0xd0, 0xd1, 0xd2, 0xd3, 0xd4, 0xd5, 0xd6, 0xd7, 0xd8, 0xd9, 0xda, 0xdb, 0xdc, 0xdd, 0xde, 0xdf, // d
+		0xe0, 0xe1, 0xe2, 0xe3, 0xe4, 0xe5, 0xe6, 0xe7, 0xe8, 0xe9, 0xea, 0xeb, 0xec, 0xed, 0xee, 0xef, // e
+		0xf0, 0xf1, 0xf2, 0xf3, 0xf4, 0xf5, 0xf6, 0xf7, 0xf8, 0xf9, 0xfa, 0xfb, 0xfc, 0xfd, 0xfe, 0xff  // f
+	};
 
-	if (_convstr) ; // Conversion succeeded.
-	else if (error->code == G_CONVERT_ERROR_NO_CONVERSION) {
-		CONV_ERROR(enc, "UTF-8");
-		// Can't convert between UTF-8 and chosen code page.
-		ExultStudio *studio = ExultStudio::get_instance();
-		studio->prompt("Failed to convert from selected codepage to UTF-8\n"
-		               "This usually happens on Windows for some ISO character sets.", "OK");
-		if (cvt_str) g_free(cvt_str);
-		return;
-	} else if (error->code == G_CONVERT_ERROR_ILLEGAL_SEQUENCE) {
-		// Need to clean string.
-		string force(str);
-		while (!_convstr && error->code == G_CONVERT_ERROR_ILLEGAL_SEQUENCE) {
-			const char illegal = force[bytes_read];
-			size_t pos;
-
-			while ((pos = force.find(illegal)) != string::npos)
-				force[pos] = '?';
-
-			CONV_ERROR(enc, "UTF-8");
-			_convstr = g_convert(force.c_str(), -1, "UTF-8", enc,
-			                     &bytes_read, &bytes_written, &error);
+	string cvt_str("");
+	if (is_7bit) {
+		cvt_str = src_str;
+		for (unsigned i = 0; i < strlen(cvt_str.c_str()); i++) {
+			cvt_str[ i ] = static_cast<char>(ToCP437[ static_cast<unsigned char>(cvt_str[ i ]) ]);
 		}
+		src_str = cvt_str.c_str();
+	}
 
-		if (!_convstr) {
-			CONV_ERROR(enc, "UTF-8");
-			// Conversion still failed; try lossy conversion.
-			const char fallback = '?';
-			_convstr = g_convert_with_fallback(force.c_str(), -1, "UTF-8", enc,
-			                                   &fallback, &bytes_read, &bytes_written, &error);
-		}
+	string tgt_str(1 + strlen(src_str)*3, ' ');
```

Comment:
Should probably be `*4` to be safe. You never know what users might be typing, and getting random crashes because someone types an emoji would not be nice.

Reply:
Also, instead of calling `strlen(src_str)` numerous times, you should call it and assign the result to a const variable.